warehouse/query-core/src/main/java/datawave/query/planner/DefaultQueryPlanner.java

```diff
@@ -1375,11 +1363,62 @@ protected ASTJexlScript processTree(final ASTJexlScript originalQueryTree, Shard
             if (log.isDebugEnabled()) {
                 log.debug("Bounded range and regex conversion has been disabled");
             }
+            
+            queryTree = checkAndExpandCompositeTerms(config, queryTree, timers, scannerFactory, metadataHelper, expansionFields);
         }
         
         return queryTree;
     }
     
+    /**
+     * Helper Function which checks whether we can utilize composite indices and then returns an updated tree.
+     * 
+     * @param config
+     *            Shard Configuration
+     * @param queryTree
+     *            incoming query tree
+     * @param timers
+     *            Query stop watch we will use to time this execution segment.
+     * @return updated query tree or the original if composite fields are disabled.
+     */
```

Comment:
I guess moving this to a function wasn't really necessary...

Reply:
It's still called 2x above, so I would say it makes sense.